src/task-actions.ts

```diff
@@ -74,6 +81,80 @@ function getFileName(filePath: string) {
     return path.basename(filePath, path.extname(filePath));
 }
 
+export async function starknetCompileCairo1Action(
+    args: TaskArguments,
+    hre: HardhatRuntimeEnvironment
+) {
+    const venvPath = hre.config.starknet.venv;
+    const manifestPath = hre.config.starknet.manifestPath || args?.manifestPath;
+    if (venvPath && !manifestPath) {
+        const msg =
+            "Could not find manifest-path\n" +
+            "The argument '—-manifest-path path/to/Cargo.toml' or manifestPath on hardhat.config.ts should be set.";
+        throw new StarknetPluginError(msg);
+    }
+
+    if (venvPath && path.basename(manifestPath) !== CARGO_FILE && venvPath) {
```

Comment:
cairo1 compiler requires the `.toml` file to be named precisely `Cargo.toml` and we check twice, if the mode is vevn, to avoid none of the errors being thrown in docker modes.

Reply:
`venvPath && ... && venvPath` still looks wrong to me

Reply:
Great catch! https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/328/commits/9ba7e4eaa5af3432772ee95439002c18e31de290